examples/ephemeral-spark/tamr-config.tf

```diff
@@ -38,25 +39,28 @@ module "tamr-config" {
   tamr_external_storage_providers = "[{'name' : 's3a_tamr_config_test','description' : 'The S3a filesystem at root of ${module.s3-data.bucket_name}','uri' : 's3a://${module.s3-data.bucket_name}/'}]"
 
   # Ephemeral Spark
-  emr_release_label           = "emr-5.29.0" # spark 2.4.4
-  emr_instance_profile_name   = module.ephemeral-spark-iam.emr_ec2_instance_profile_name
-  emr_service_role_name       = module.ephemeral-spark-iam.emr_service_role_name
-  emr_key_pair_name           = module.emr_key_pair.key_pair_key_name
-  emr_subnet_id               = var.ec2_subnet_id
-  master_instance_type        = "m4.large"
-  master_ebs_volumes_count    = 1
-  master_ebs_size             = 50
-  master_ebs_type             = "gp2"
-  core_ebs_volumes_count      = 1
-  core_ebs_size               = 200
-  core_ebs_type               = "gp2"
-  core_group_instance_count   = 4
-  core_instance_type          = "r5.4xlarge"
-  emr_service_access_sg_id    = module.ephemeral-spark-sgs.emr_service_access_sg_id
-  emr_managed_master_sg_id    = module.ephemeral-spark-sgs.emr_managed_master_sg_id
-  emr_additional_master_sg_id = module.ephemeral-spark-sgs.emr_additional_master_sg_id
-  emr_managed_core_sg_id      = module.ephemeral-spark-sgs.emr_managed_core_sg_id
-  emr_additional_core_sg_id   = module.ephemeral-spark-sgs.emr_additional_core_sg_id
+  emr_release_label         = "emr-5.29.0" # spark 2.4.4
+  emr_instance_profile_name = module.ephemeral-spark-iam.emr_ec2_instance_profile_name
+  emr_service_role_name     = module.ephemeral-spark-iam.emr_service_role_name
+  emr_key_pair_name         = module.emr_key_pair.key_pair_key_name
+  emr_subnet_id             = var.ec2_subnet_id
+  master_instance_type      = "m4.large"
+  master_ebs_volumes_count  = 1
+  master_ebs_size           = 50
+  master_ebs_type           = "gp2"
+  core_ebs_volumes_count    = 1
+  core_ebs_size             = 200
+  core_ebs_type             = "gp2"
+  core_group_instance_count = 4
+  core_instance_type        = "r5.4xlarge"
```

Comment:
Given the modest spark settings in the "spark" section above (combined with the defaults for the unspecified fields here), we could reduce this instance type by quite a lot, since it is much larger than we would need.
```suggestion
  core_instance_type        = "r5.xlarge"
```

I think what makes this unclear is that we are relying on defaults that we don't see here.

Here's a snippet from a recent internal test deployment that is more complete:
```
  # Spark
  spark_emr_cluster_id           = ""
  spark_cluster_log_uri          = "s3n://${module.s3-logs.bucket_name}/${var.path_to_spark_logs}"
  spark_driver_memory            = "5G"
  spark_executor_instances       = 15
  spark_executor_memory          = "6G"
  spark_executor_cores           = 1
  tamr_data_path                 = "tamr/unify-data"
  tamr_spark_config_override     = "[{'name' : 'sparkOverride1','executorInstances' : '2','sparkProps' : {'spark.cores.max' : '4'}},{'name' : 'sparkOverride2','driverMemory' : '4G','executorMemory' : '5G'}]"
  tamr_spark_properties_override = "{'spark.driver.maxResultSize':'4g'}"

  # Ephemeral Spark
  emr_release_label           = "emr-5.29.0" # spark 2.4.4
  emr_instance_profile_name   = module.ephemeral-spark-iam.emr_ec2_instance_profile_name
  emr_service_role_name       = module.ephemeral-spark-iam.emr_service_role_name
  emr_key_pair_name           = module.emr_key_pair.key_pair_key_name
  emr_subnet_id               = var.ec2_subnet_id
  master_instance_type        = "m4.large"
  master_ebs_volumes_count    = 1
  master_ebs_size             = 50
  master_ebs_type             = "gp2"
  core_ebs_volumes_count      = 1
  core_ebs_size               = 200
  core_ebs_type               = "gp2"
  core_group_instance_count   = 4
  core_instance_type          = "r5.xlarge"
```

Reply:
Yes, haha, I agree... Thanks for the example! I'll update the example.